lib/inject.rb

```diff
@@ -1,23 +1,38 @@
 # Return true if all elements are equal to the argument, false otherwise.
 def all_equal?(argument, elements)
   # TODO: rewrite to use inject instead of recursion
-  case elements
-  when []
-    true
-  else
-    first, *rest = elements
-    first == argument && all_equal?(argument, rest)
-  end
+  #case elements
+  #when []
+    #true
+  #else
+    #first, *rest = elements
+    #first == argument && all_equal?(argument, rest)
+  #end
+
+  # This also works, but is more verbose
+  #elements.inject(true) do |result, n|
+    #result && n == argument
+  #end
+
+  elements.inject(:&) == argument
```

Comment:
I don't think this would work, except for limited cases. Did they give you a test case? For example:

```
[1] pry(main)> [1,2,3,4,5].inject(:&)
=> 0
```

But I don't think `all_equal([1,2,3,4,5], 0)` is the intended usage. Instead, take a look at the ruby method `Enumerable#all?`, which semantically matches what we're doing here:

```
[3] pry(main)> [2, 4, 6, 8, 10].all? { |e| e.even? }
=> true
```